### ISSUE TITLE
Always encrypted numeric column is considered as decimal

### ISSUE THREAD

**Issue by Unnamed1984** [dotnet/efcore#25932]
Hi.
I encountered an exception while I was trying to work with encrypted columns via EF Core:

SqlException: Operand type clash: decimal(8,6) encrypted with (encryption_type = 'DETERMINISTIC', encryption_algorithm_name = 'AEAD_AES_256_CBC_HMAC_SHA_256', column_encryption_key_name = 'My_Column_Encryption_Key_1', column_encryption_key_database_name = 'mydb') is incompatible with numeric(18,6) encrypted with (encryption_type = 'DETERMINISTIC', encryption_algorithm_name = 'AEAD_AES_256_CBC_HMAC_SHA_256', column_encryption_key_name = 'My_Column_Encryption_Key_1', column_encryption_key_database_name = 'mydb')

An encrypted column has type numeric(18,6) in a database.
I used this configuration in my code:
```
string decimal MyProperty { get; set; }
...
modelBuilder.Property(p => p.MyProperty).HasColumnType("numeric(16,8)");
```
Calling HasColumnType method fixed the similar issue for varchar columns, doesn't work for numeric though.

EF Core version: 3.1.12
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET Core 3.1
Operating system: Windows
IDE: Visual Studio 2019
The approach used is DB First.

Full stack trace is here:
"   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.Execute(IRelationalConnection connection)\r\n   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.Execute(IEnumerable`1 commandBatches, IRelationalConnection connection)\r\n   at Microsoft.EntityFrameworkCore.Storage.RelationalDatabase.SaveChanges(IList`1 entries)\r\n   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(IList`1 entriesToSave)\r\n   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(DbContext _, Boolean acceptAllChangesOnSuccess)\r\n   at Microsoft.EntityFrameworkCore.Storage.ExecutionStrategy.ExecuteImplementation[TState,TResult](Func`3 operation, Func`3 verifySucceeded, TState state)\r\n   at Microsoft.EntityFrameworkCore.Storage.ExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)\r\n   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(Boolean acceptAllChangesOnSuccess)\r\n   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges(Boolean acceptAllChangesOnSuccess)\r\n

**Comment by Unnamed1984** [comment]
Here is the workaround:
[https://stackoverflow.com/questions/48993337/ef-core-decimal-precision-for-always-encrypted-column/69129324#69129324](url)

But still, the issue should be fixed.

**Comment by ajcvickers** [comment]
@Unnamed1984 Please test this on the latest [daily build](https://github.com/dotnet/efcore/blob/release/6.0/docs/DailyBuilds.md). We have fixed several issues in this area for the 6.0 release.

Alternately, please attach a small, runnable project or post a small, runnable code listing that reproduces what you are seeing and I will try on the daily build.

**Comment by ajcvickers** [comment]
**EF Team Triage:** Closing this issue as the requested additional details have not been provided and we have been unable to reproduce it.

*BTW this is a canned response and may have info or details that do not directly apply to this particular issue. While we'd like to spend the time to uniquely address every incoming issue, we get a lot traffic on the EF projects and that is not practical. To ensure we maximize the time we have to work on fixing bugs, implementing new features, etc. we use canned responses for common triage decisions.*
